src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -11,6 +16,12 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIProxy
     {
+        private static readonly char[] s_semicolonSeparator = { ';' };
```

Comment:
Why do these need to be char arrays? Split can accept a single char as an argument. Although I'm not sure if that conflicts with the multiple allocation comment made in the earlier PR.